src/Hit/Issue.hs

```diff
@@ -121,6 +123,44 @@ createIssue title login = withOwnerRepo $ \token owner repo -> case token of
         errorMessage errorText
         pure $ Left $ ParseError errorText
 
+{- | Assign the user to the given 'Issue'.
+
+This function can fail assignment due to the following reasons:
+
+ * Auth token fetch failure
+ * Assignment query to GutHub failure
+
+The function should inform user about corresponding 'Error' in each case and
+continue working.
+-}
+assignIssue :: Issue -> Text -> IO ()
+assignIssue issue username = do
+    res <- withOwnerRepo $ \token owner repo -> case token of
+        Just auth -> do
+            let assignee :: Name User
+                assignee = makeName @User username
+            let curAssignees :: V.Vector (Name User)
+                curAssignees = V.map simpleUserLogin $ issueAssignees issue
+
+            if V.elem assignee curAssignees
+            then pure $ Right (issue, True)
+            else do
+                -- TODO: this is hack to cheat on GitHub library, as it
+                -- doesn't use the correct id in query.
+                let issId = mkIssueId (unIssueNumber $ issueNumber issue)
+                (, False) <<$>> GitHub.editIssue auth owner repo issId editOfIssue
+                    { editIssueAssignees = Just $ V.cons assignee curAssignees
+                    }
+
+        Nothing -> pure $ Left $ UserError "Can not get the GITHUB_TOKEN to assign you to the issue."
+    case res of
+        Right (iss, False) -> successMessage $ "You were assigned to the issue #" <>
+            show (unIssueNumber $ issueNumber iss)
+        Right (_iss, True) -> pass
```

Comment:
Do you think it would be more readable to rewrite `case` to:

```haskell
Right (iss, isAlreadyAssigned) ->
    if isAlreadyAssigned
    then pass
    else ...
```